### PR TITLE
Fix build error when using git-submodule

### PR DESCRIPTION
**Pull request by k-takata**
This is an additional fix for #1475.
When git-submodule is used, an error is occurred because `.git/*` doesn't exist.
Just use `.git` instead.

### CODE REVIEW

**Comment by k-takata** [comment]
Or is it better always run `misc/gen-repoinfo` when running `make`?

**Comment by masatake** [comment]
How about following approach:
````
 diff --git a/Makefile.am b/Makefile.am
index 825c05e..e854b11 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -78,8 +78,13 @@ CLEANFILES = $(REPOINFO_HEADS)
 $(REPOINFO_OBJS): $(REPOINFO_SRCS) $(REPOINFO_HEADS)
 if BUILD_IN_GIT_REPO
 GEN_REPOINFO = $(srcdir)/misc/gen-repoinfo
+if BUILD_AS_GIT_SUBMODULE
+$(REPOINFO_HEADS): $(GEN_REPOINFO) $(srcdir)/.git
+       $(GEN_REPOINFO) > $@
+else
 $(REPOINFO_HEADS): $(GEN_REPOINFO) $(srcdir)/.git/*
        $(GEN_REPOINFO) > $@
+fi
 else
 $(REPOINFO_HEADS):
        echo > $@
diff --git a/configure.ac b/configure.ac
index 50b6595..715f7d7 100644
--- a/configure.ac
+++ b/configure.ac
@@ -310,17 +310,28 @@ AM_CONDITIONAL([HAVE_RST2MAN], [test "x$RST2MAN" != "xno"])
 # Checks for operating environment
 # --------------------------------
 in_git_repo=no
+as_git_submodule=no
 AC_MSG_CHECKING(building in a git repository)
 if test -f "${srcdir}/.git/HEAD" || test -f "${srcdir}/.git"; then
        in_git_repo=yes
 fi
 AC_MSG_RESULT(${in_git_repo})
 
+if test "${in_git_repo}" = yes; then
+       AC_MSG_CHECKING(building as a git submodule)
+       if test -f "${srcdir}/.git"; then
+               as_git_submodule=yes
+       fi
+       AC_MSG_RESULT(${as_git_submodule})
+fi
+
 AC_CHECK_PROGS([GIT], [git])
 if ! test "${GIT+set}" = "set"; then
        in_git_repo=no
+       as_git_submodule=no
 fi
 AM_CONDITIONAL([BUILD_IN_GIT_REPO], [test "x${in_git_repo}" = "xyes"])
+AM_CONDITIONAL([BUILD_AS_GIT_SUBMODULE], [test "x${as_git_submodule}" = "xyes"])
```

I didn't test the change but you may know my intention.

**Comment by k-takata** [comment]
`repoinfo.h` might not be properly updated with the approach, because `.git` is not updated even new commits are added to the repository.
I added a new commit to this PR.

File: misc/gen-repoinfo [hunk 1]
```diff
@@ -1,4 +1,21 @@
...
+	oldinfo=`cat $rinfoheader`
+else
+	echo > $rinfoheader
+fi
+
 rinfo=`git describe --tag --exact-match HEAD 2> /dev/null || git rev-parse --short HEAD`
-echo "#define CTAGS_REPOINFO \"${rinfo}\""
+newinfo="#define CTAGS_REPOINFO \"${rinfo}\""
 
+if [ "$newinfo" != "$oldinfo" ]; then
```
Comment (masatake) [inline misc/gen-repoinfo]: neat technique.

**Pull request merged**